extensions/wikia/InsightsV2/InsightsHelper.php

```diff
@@ -0,0 +1,200 @@
+	 * 'insights-notification-message-fixed-uncategorizedpages',
+	 * 'insights-notification-message-fixed-wantedpages'
+	 * 'insights-notification-message-fixed-withoutimages',
+	 */
+	const INSIGHT_FIXED_MSG_PREFIX = 'insights-notification-message-fixed-';
+
+	private static $defaultInsights = [
+		InsightsUncategorizedModel::INSIGHT_TYPE	=> 'InsightsUncategorizedModel',
+		InsightsWithoutimagesModel::INSIGHT_TYPE	=> 'InsightsWithoutimagesModel',
+		InsightsDeadendModel::INSIGHT_TYPE			=> 'InsightsDeadendModel',
+		InsightsWantedpagesModel::INSIGHT_TYPE		=> 'InsightsWantedpagesModel'
+	];
+
+
+	/**
+	 * Prepare array with all available insights pages
+	 *
+	 * @return array
+	 */
+	public static function getInsightsPages() {
+		global $wgEnableInsightsInfoboxes, $wgEnableFlagsExt, $wgEnableTemplateClassificationExt,
+			   $wgEnableInsightsPagesWithoutInfobox, $wgEnableInsightsTemplatesWithoutType;
+
+		/* Order of inserting determines default order on insights entry points list */
+		$dynamicInsights = [];
+
+		/* Add TemplatesWithoutType insight */
+		if ( !empty( $wgEnableTemplateClassificationExt ) && !empty( $wgEnableInsightsTemplatesWithoutType ) ) {
+			$dynamicInsights[InsightsTemplatesWithoutTypeModel::INSIGHT_TYPE] = 'InsightsTemplatesWithoutTypeModel';
+		}
+
+		/* Add Infoboxes insight */
+		if ( !empty( $wgEnableInsightsInfoboxes ) ) {
+			$dynamicInsights[InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE] = 'InsightsUnconvertedInfoboxesModel';
+		}
+
+		/* Add PagesWithoutInfobox insight */
+		if ( !empty( $wgEnableTemplateClassificationExt ) && !empty( $wgEnableInsightsPagesWithoutInfobox ) ) {
+			$dynamicInsights[InsightsPagesWithoutInfoboxModel::INSIGHT_TYPE] = 'InsightsPagesWithoutInfoboxModel';
+		}
+
+		/* Add Flags insight */
+		if ( !empty( $wgEnableFlagsExt ) ) {
+			$dynamicInsights[InsightsFlagsModel::INSIGHT_TYPE] = 'InsightsFlagsModel';
+		}
+
+
+		return array_merge( $dynamicInsights, self::$defaultInsights );
+	}
+
+	/**
+	 * Get list of insights which should be highlighted in insights list (should have red dot)
+	 *
+	 * @return array
+	 */
+	public static function getHighlightedInsights() {
+		global $wgEnableTemplateClassificationExt, $wgEnableInsightsInfoboxes, $wgEnableInsightsTemplatesWithoutType;
+
+		$highlightedInsights = [];
+
+		if ( !empty( $wgEnableInsightsInfoboxes ) ) {
+			$highlightedInsights[] = InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE;
+		}
+
+		if ( !empty( $wgEnableTemplateClassificationExt ) && !empty( $wgEnableInsightsTemplatesWithoutType ) ) {
+			$highlightedInsights[] = InsightsTemplatesWithoutTypeModel::INSIGHT_TYPE;
+		}
+
+		return $highlightedInsights;
+	}
+
+	/**
+	 * Returns a full URL for a known subpage and a NULL for an unknown one.
+	 * @param string $subpage slug of subpage
+	 * @param array $params params
+	 * @return string|null
+	 */
+	public static function getSubpageLocalUrl( $subpage = false, Array $params = [] ) {
```

Comment:
Default value for ``$subpage`` doesn't make any sense here. Calling ``getSubpageLocalUrl`` without subpage parameter returns just null, so each usage of this method should provide a ``$subpage`` param.